### PR TITLE
Fix Email Reset 403 Error

### PR DESCRIPTION
**Pull request by SudharakaP**
Fixes the 403 error that appears at the top when going into the email reset screen. I believe that this is due to some leftover code that we have to remove when upgrading. 

Related to https://github.com/jhipster/jhipster-online/pull/170 and Fixes https://github.com/jhipster/jhipster-online/issues/187

cc: @vishal423

### CODE REVIEW

**Comment by vishal423** [comment]
@SudharakaP, I am not very confident on this scenario. Have you tested with `jhispter.mail.enabled` `true/false` values?

**Comment by SudharakaP** [comment]
@vishal423 : Yes I did test this. Here's a screen-cast. I also realized I've mistakenly removed warning message for non-registered email which I've added back; that should be there I think to give the user a warning if they try to enter a non-registered email. 

Peek 2020-06-06 06-40

**Comment by SudharakaP** [comment]
Actually my original commit was correct we need to remove the functionality of the non-registered email waning message; but I haven't done it fully in my original commit. So I've went ahead and made some changes such that now even when a user enters a non-existent email address we just say that the password reset request is successful. This is in agreement with the changes on the latest jhispter-generator; basically I mimicked the code there. :smile: 

If you have questions or issues let me know. I think this PR is now ready to review. :smile:

File: src/main/java/io/github/jhipster/online/web/rest/AccountResource.java [hunk 1]
```diff
@@ -186,19 +185,20 @@ public void changePassword(@RequestBody PasswordChangeDTO passwordChangeDto) {
...
+            // but log that an invalid attempt has been made
+            log.warn("Password reset requested for non existing mail '{}'", mail);
         }
     }
 
     @PostMapping(path = "/account/reset-password/link")
     public String requestPasswordResetAndReturnLink(@RequestBody String mail) {
-        User user = userService.requestPasswordReset(mail).orElseThrow(EmailNotFoundException::new);
-        return userService.generatePasswordResetLink(user.getResetKey());
+        Optional<User> user = userService.requestPasswordReset(mail);
```
Comment (vishal423) [inline src/main/java/io/github/jhipster/online/web/rest/AccountResource.java:200]: optional can be empty here

File: src/main/java/io/github/jhipster/online/web/rest/AccountResource.java [hunk 2]
```diff
@@ -186,19 +185,20 @@ public void changePassword(@RequestBody PasswordChangeDTO passwordChangeDto) {
      */
     @PostMapping(path = "/account/reset-password/init")
     public void requestPasswordReset(@RequestBody String mail) {
-        if (mailService.isEnabled()) {
```
Comment (vishal423) [inline src/main/java/io/github/jhipster/online/web/rest/AccountResource.java:189]: where do we handle it now ?

**Comment by vishal423** [comment]
@SudharakaP, to be honest, I still don't understand how do you handle disabled email flow. Please note that there is a deviation on it's handing with the main generator. If you look into git history, then, those changes were added on top of the generated code. Before we remove, we should get context on their usage.

File: src/main/java/io/github/jhipster/online/web/rest/AccountResource.java (hunk 1 above)
Comment (SudharakaP) [reply src/main/java/io/github/jhipster/online/web/rest/AccountResource.java:200]: Ah I missed that. Corrected it. :+1:

**Comment by SudharakaP** [comment]
@vishal423 : I think they were added on top of the generated code as part of; https://github.com/jhipster/generator-jhipster/pull/7755. Then in the main generator it was not merged and the [associated code in jhipster-framework is reverted](https://github.com/jhipster/generator-jhipster/pull/7755#issuecomment-467474968). Now if we generate a new project and set, `jhipster.mail.enabled` to `false` there's no disabling that happens; so there's no disabled email flow; it's always enabled. Unless we recreate, https://github.com/jhipster/generator-jhipster/pull/7755 support I think the `jhipster.mail.enabled` property is not meaningful in any way. This was my understanding and that is why I tried to remove this disable email flow. 

I can be wrong here; please correct me if I am. :smile:

File: src/main/java/io/github/jhipster/online/web/rest/AccountResource.java (hunk 2 above)
Comment (SudharakaP) [reply src/main/java/io/github/jhipster/online/web/rest/AccountResource.java:189]: Please [see my below comment](https://github.com/jhipster/jhipster-online/pull/175#issuecomment-640106599) and please do correct me if I am wrong. :smile:

**Comment by pascalgrimaud** [comment]
@jdubois : I remember something about email, to be activated or not... Do you remember ? Is this PR correct ?

**Pull request merged**